src/main/java/org/opensearch/neuralsearch/processor/chunker/FixedTokenLengthChunker.java

```diff
@@ -0,0 +1,169 @@
+    public static final String TOKENIZER_FIELD = "tokenizer";
+
+    // default values for each parameter
+    private static final int DEFAULT_TOKEN_LIMIT = 500;
+    private static final BigDecimal DEFAULT_OVERLAP_RATE = new BigDecimal("0");
+    private static final int DEFAULT_MAX_TOKEN_COUNT = 10000;
+    private static final String DEFAULT_TOKENIZER = "standard";
+
+    private static final BigDecimal OVERLAP_RATE_UPPER_BOUND = new BigDecimal("0.5");
+
+    private final AnalysisRegistry analysisRegistry;
+
+    public FixedTokenLengthChunker(AnalysisRegistry analysisRegistry) {
+        this.analysisRegistry = analysisRegistry;
+    }
+
+    private void validatePositiveIntegerParameter(Map<String, Object> parameters, String fieldName) {
+        // this method validate that parameter is a positive integer
+        if (!parameters.containsKey(fieldName)) {
+            // all parameters are optional
+            return;
+        }
+        if (!(parameters.get(fieldName) instanceof Number)) {
+            throw new IllegalArgumentException(
+                "fixed length parameter [" + fieldName + "] cannot be cast to [" + Number.class.getName() + "]"
+            );
+        }
+        if (((Number) parameters.get(fieldName)).intValue() <= 0) {
+            throw new IllegalArgumentException("fixed length parameter [" + fieldName + "] must be positive");
+        }
+    }
+
+    private List<String> tokenize(String content, String tokenizer, int maxTokenCount) {
+        AnalyzeAction.Request analyzeRequest = new AnalyzeAction.Request();
+        analyzeRequest.text(content);
+        analyzeRequest.tokenizer(tokenizer);
+        try {
+            AnalyzeAction.Response analyzeResponse = analyze(analyzeRequest, analysisRegistry, null, maxTokenCount);
+            return analyzeResponse.getTokens().stream().map(AnalyzeAction.AnalyzeToken::getTerm).collect(Collectors.toList());
+        } catch (IOException e) {
+            throw new RuntimeException("Fixed token length algorithm meet with exception: " + e);
+        }
+    }
+
+    /**
+     * Validate the chunked passages for fixed token length algorithm
+     *
+     * @param parameters a map containing parameters, containing the following parameters
+     * 1. tokenizer the analyzer tokenizer in opensearch, please check https://opensearch.org/docs/latest/analyzers/tokenizers/index/
+     * 2. token_limit the token limit for each chunked passage
+     * 3. overlap_rate the overlapping degree for each chunked passage, indicating how many token comes from the previous passage
+     * 4. max_token_count the max token limit for the tokenizer
+     * @throws IllegalArgumentException If max_token_count and token_limit is not a positive integer
+     * @throws IllegalArgumentException If overlap_rate is not within range [0, 0.5]
+     * @throws IllegalArgumentException If tokenizer is not a string
+     */
+    @Override
+    public void validateParameters(Map<String, Object> parameters) {
```

Comment:
Same as delimiter chunk, if we can extract all the parameters used in class level, then we can remove a lot of retrieval codes.

Reply:
How about adding a new method to the interface named `parseParameters`? This method will parse the `Map<String, Object> parameters` into member variable within the class.

Reply:
Class level parameter has been implemented. Please help review.